conda_smithy/configure_feedstock.py

```diff
@@ -102,10 +105,10 @@ def render_circle(jinja_env, forge_config, forge_dir):
 
 @contextmanager
 def fudge_subdir(subdir):
-    orig = conda_build.metadata.cc.subdir
-    conda_build.metadata.cc.subdir = subdir
+    orig = cc.subdir
+    cc.subdir = subdir
     yield
-    conda_build.metadata.cc.subdir = orig
+    cc.subdir = orig
```

Comment:
My opinion on why this is going wrong is that cc is a global variable.  Somewhere in that conda_interface, we're breaking the link to conda, so that there are actually 2 instances floating around.  You are current changing the one in conda-build, but that does not seem to affect the one in conda.